Client/Frontend/Browser/Onboarding/OnboardingNavigationController.swift

```diff
@@ -55,7 +56,20 @@ class OnboardingNavigationController: UINavigationController {
                     if progress == .rewards || progress == .ads {
                         return BraveAds.isCurrentLocaleSupported() ? [.adsCountdown] : [.rewardsAgreement]
                     }
-                    return BraveAds.isCurrentLocaleSupported() ? [.searchEnginePicker, .shieldsInfo, .rewardsAgreement, .adsCountdown] : [.searchEnginePicker, .shieldsInfo, .rewardsAgreement]
+                    
+                    var newUserScreens: [Screens] = [.searchEnginePicker, .shieldsInfo, .rewardsAgreement]
+                    
+                    if BraveAds.isCurrentLocaleSupported() {
+                        newUserScreens.append(.adsCountdown)
+                    }
+                    
+                    // FIXME: Update to iOS14 clipboard api once ready (#2838)
+                    if Preferences.URP.referralLookupOutstanding.value == true
```

Comment:
@iccub - As I understand, the `privacyConsent` screen will be displayed multiple times if referral lookup fails but the code is read from the clipboard. I think it might make sense to assign `URP.referralCode.value` once the clipboard is accessed and then display the privacy consent screen only when `URP.referralCode.value == null`. Let me know if I'm missing something.

Reply:
Good idea